system-tests/azure-blob-transfer-tests-non-provisioning/src/test/java/org/eclipse/edc/test/system/blob/PreProvisionedTransferIntegrationTest.java

```diff
@@ -0,0 +1,142 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *       ZF Friedrichshafen AG - add management api configurations
+ *       Fraunhofer Institute for Software and Systems Engineering - added IDS API context
+ *
+ */
+
+package org.eclipse.edc.test.system.blob;
+
+import com.azure.core.util.BinaryData;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.edc.azure.testfixtures.AbstractAzureBlobTest;
+import org.eclipse.edc.azure.testfixtures.TestFunctions;
+import org.eclipse.edc.azure.testfixtures.annotations.AzureStorageIntegrationTest;
+import org.eclipse.edc.connector.transfer.spi.types.TransferProcessStates;
+import org.eclipse.edc.junit.extensions.EdcRuntimeExtension;
+import org.eclipse.edc.spi.security.Vault;
+import org.eclipse.edc.test.system.utils.Participant;
+import org.eclipse.edc.test.system.utils.PolicyFixtures;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+import java.net.URI;
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static java.lang.String.format;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.eclipse.edc.azure.blob.AzureBlobStoreSchema.CORRELATION_ID;
+import static org.eclipse.edc.connector.transfer.spi.types.TransferProcessStates.COMPLETED;
+import static org.eclipse.edc.test.system.blob.Constants.POLL_INTERVAL;
+import static org.eclipse.edc.test.system.blob.Constants.TIMEOUT;
+import static org.eclipse.edc.test.system.blob.ConsumerConstants.CONSUMER_PROPERTIES;
+import static org.eclipse.edc.test.system.blob.ProviderConstants.ASSET_FILE;
+import static org.eclipse.edc.test.system.blob.ProviderConstants.BLOB_CONTENT;
+import static org.eclipse.edc.test.system.blob.ProviderConstants.PROVIDER_PROPERTIES;
+import static org.testcontainers.shaded.org.awaitility.Awaitility.await;
+
+@Testcontainers
+@AzureStorageIntegrationTest
+public class PreProvisionedTransferIntegrationTest extends AbstractAzureBlobTest {
```

Comment:
why do we need this to be in a separate module? Could the test class not just be in the `azure-blob-transfer-tests` module?

Reply:
Unfortunately that's, as of my understanding, not possible, as the classpath needs to be different. Basically it's one reason why I think this test case make sense, as it documents this aspect. That was about that the data destination properties are replaced as a whole when a provisioning step is applied, so having the provisioning extension on the classpath will cause this integration test to fail.

Reply:
the classpath of the runtime is determined by `EdcRuntimeExtension`'s first argument. no need to put the test itself on a separate module too.
Generally though, things like that **MUST** be documented, even in tests.